playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml

```diff
@@ -229,3 +229,56 @@
   tasks:
   - include: docker/upgrade.yml
     when: l_docker_upgrade is defined and l_docker_upgrade | bool and not openshift.common.is_atomic | bool
+
+- name: Drain and upgrade master nodes
+  hosts: oo_masters_to_config:&oo_nodes_to_upgrade
```

Comment:
oo_nodes_to_upgrade is calculated by processing openshift_upgrade_nodes_serial, and openshift_upgrade_nodes_label, both I think would be an error to apply here as you should never be able to trigger this for more than one host at a time in parallel, or skip some masters due to labels. IMO it would be a bug to process these here. 

You could however use the node group which does *not* apply those variables, which is I believe oo_nodes_to_config, that would give you the clarity but not include any processing of the label var. (and go to serial: 1 as well)